cmd/restic/cmd_sync.go

```diff
@@ -269,9 +282,67 @@ func runSync(opts SyncOptions, gopts GlobalOptions, term *termstatus.Terminal, a
 	if totalErrors > 0 {
 		Printf("There were %d errors\n", totalErrors)
 	}
+
 	p.Finish(id)
-	p.V("Files To Get: %v", fileTracker.FilesToGet)
-	p.V("Files To Delete: %v", fileTracker.FilesToDelete)
+
+	if globalOptions.verbosity >= 1 {
+		var filesFetched []string
+		for k, v := range fileTracker.FilesToGet {
+			if v == true {
+				filesFetched = append(filesFetched, k)
+			}
+		}
+		Verbosef("Files Fetched: %v\n", filesFetched)
+		Verbosef("Files To Delete: %v\n", fileTracker.FilesToDelete)
+		Verbosef("Directories To Delete: %v\n", fileTracker.UnexpectedDirs)
+	}
+
+	if opts.Delete {
+		errCount := deleteFilesAndDirectories(fileTracker, opts)
+		Printf("%d errors while deleting files\n", errCount)
+	} else {
+		Verbosef("Skipping file deletion as --delete was not passed.")
+	}
 
 	return err
 }
+
+func deleteFilesAndDirectories(fileTracker FileTracker, opts SyncOptions) int {
+	errorCount := 0
+	for _, file := range fileTracker.FilesToDelete {
+		fileName := path.Join(opts.Target, file)
+		Verbosef("Deleting %s\n", fileName)
+		err := os.Remove(fileName)
+		if err != nil {
+			errorCount++
+			Printf("Error deleting %s\n", fileName)
+		}
+	}
+
+	// As we are not using RemoveAll (hence why we do files first), we cannot delete non-empty directories, so a bug in
+	// the directory marking logic cannot result in the deletion of directories containing files
+	// We must delete deeper directories first, however - e.g. in the case of /A/B/, we must first delete B/, then A/
+	depthSortedUnexpectedDirs := map[int][]string{}
```

Comment:
This is a nice implementation of deleting our paths with the most components first.

Reply:
I'm a little scared there may be some crazy edge case if e.g. directory symlinks are traversed, but I figure that shouldn't be the case. Worst case we'll just have a few leftover empty directories with errors reported in the update log